### ISSUE TITLE
Horizon database growth rates

### ISSUE THREAD

**Issue by Shaptic** [stellar/go#3752]
This issue exists is purely for tracking this task.

#### Goal
Identify the effects of transaction submission on Horizon database size.

#### Tasks
- [x] Identify how much the database grows after certain numbers of transactions submissions.
- [x] Identify whether or not the database grows differently depending on successful or failed transactions, as well as factors such as:
  * [x] varying operation counts
  * [x] varying failing operations
  * [ ] varying signature counts
 - [x] Identify which tables grow (and by how much) for the above scenarios

**Note:** Scale beyond this is unrealistic, because with ~5s ledger close times testing e.g. 10000 transactions means ~13 hours of txsub.

#### Relevant Info

 - DB tables: `history_transactions` and friends
 - [Getting DB sizes](https://stackoverflow.com/a/18910638) in PostgreSQL
 - Transactions [glossary page](https://developers.stellar.org/docs/glossary/transactions)
 - Transactions [API object](https://developers.stellar.org/api/resources/transactions/object/)

**Comment by Shaptic** [comment]
Results are available [here](https://docs.google.com/document/d/1hrQyPwgtBWECnjOZGAYXj2qKYZ4fO6nALTes5TUdn1o/view). 

**TL;DR:** Depending on your position on the optimist <-----> pessimist scale, **the Horizon database will grow by anywhere between 2 and 20 terabytes** by the end of the year. 

This scale varies based on your subjective prediction on the popularity of our network. The lower end estimates this with a +3 tx/ledger growth rate, which follows a linear trend line since 2019. The higher end estimates this with a +23 tx/ledger growth rate, which follows a linear trend line since the start of 2021. Obviously, the past few months have been anomalous in the crypto space (and the uptick is slowing down significantly), but it's important to consider a range of possibilities.

You can refer to [this plot](https://www.wolframalpha.com/input/?i=plot+%282160000x*236*%280.223*2.57+%2B+0.777*2.25%29%29%2F%281024%5E2%29+from+x%3D3+to+23) to see growth amounts (in GB) based on different growth rates.
